src/test/java/org/cryptomator/common/mount/MountWithinParentUtilTest.java

```diff
@@ -0,0 +1,244 @@
+		Files.createDirectory(mount);
+
+		prepareParentNoMountPoint(mount);
+		assertTrue(Files.notExists(mount, NOFOLLOW_LINKS));
+
+		assumeTrue(SystemUtils.IS_OS_WINDOWS);
+		assertTrue(isHidden(hideaway));
+	}
+
+	@Test
+	void testHandleMountPointFolderDoesNotExist(@TempDir Path parentDir) throws IOException {
+		assertFalse(handleMountPointFolder(parentDir.resolve("notExisting")));
+	}
+
+	@Test
+	void testHandleMountPointFolderIsFile(@TempDir Path parentDir) throws IOException {
+		var regularFile = parentDir.resolve("regularFile");
+		Files.createFile(regularFile);
+
+		assertThrows(MountPointNotEmptyDirectoryException.class, () -> {
+			handleMountPointFolder(regularFile);
+		});
+		assertTrue(Files.exists(regularFile));
+	}
+
+	@Test
+	void testHandleMountPointFolderIsNotEmpty(@TempDir Path parentDir) throws IOException {
+		var regularFolder = parentDir.resolve("regularFolder");
+		var dummyFile = regularFolder.resolve("dummy");
+		Files.createDirectory(regularFolder);
+		Files.createFile(dummyFile);
+
+		assertThrows(MountPointNotEmptyDirectoryException.class, () -> {
+			handleMountPointFolder(regularFolder);
+		});
+		assertTrue(Files.exists(regularFolder));
+		assertTrue(Files.exists(dummyFile));
+	}
+
+	@Test
+	void testHandleMountPointFolder(@TempDir Path parentDir) throws IOException {
+		//Sadly can't easily create files with "Other" attribute
+		var regularFolder = parentDir.resolve("regularFolder");
+		Files.createDirectory(regularFolder);
+
+		assertTrue(handleMountPointFolder(regularFolder));
+		assertTrue(Files.exists(regularFolder));
+	}
+
+	@Test
+	void testRemoveResidualHideawayFile(@TempDir Path parentDir) throws IOException {
+		var hideaway = parentDir.resolve("hideaway");
+		Files.createFile(hideaway);
+
+		assertThrows(HideawayNotDirectoryException.class, () -> removeResidualHideaway(parentDir.resolve("mount"), hideaway));
+	}
+
+	@Test
+	void testRemoveResidualHideawayNotEmpty(@TempDir Path parentDir) throws IOException {
+		var hideaway = parentDir.resolve("hideaway");
+		Files.createDirectory(hideaway);
+		Files.createFile(hideaway.resolve("dummy"));
+
+		assertThrows(DirectoryNotEmptyException.class, () -> removeResidualHideaway(parentDir.resolve("mount"), hideaway));
+	}
+
+	@Test
+	void testCleanupNoHideaway(@TempDir Path parentDir) {
+		assertDoesNotThrow(() -> cleanup(parentDir.resolve("mount")));
+	}
+
+	@Test
+	void testCleanup(@TempDir Path parentDir) throws IOException {
+		var mount = parentDir.resolve("mount");
+		var hideaway = getHideaway(mount);
+		Files.createDirectory(hideaway);
+
+		cleanup(mount);
+		assertTrue(Files.exists(mount, NOFOLLOW_LINKS));
+		assertTrue(Files.notExists(hideaway, NOFOLLOW_LINKS));
+		assertFalse(isHidden(mount));
+	}
+
+	@Test
+	@EnabledOnOs(OS.WINDOWS)
+	void testGetHideawayRootDirWin() {
+		var mount = Path.of("C:\\mount");
+		var hideaway = getHideaway(mount);
+
+		assertEquals(mount.getParent().toAbsolutePath(), Path.of("C:\\").toAbsolutePath());
```

Comment:
This is nothing we have to check, because it is covered by the JDK.

Reply:
Resolved by 1c34402c87077252871cea2cbf4ec5f546adcb4d.